### ISSUE TITLE
Nothing loads

### ISSUE THREAD

**Issue by bbaster** [Docile-Alligator/Infinity-For-Reddit#1648]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only occurs with a certain link, post, image..., I will include the URL.

### App version

7.2.1

### Where did you get the app from

Other

### Android version

ArrowOS 13.1

### Device model

Poco X3 Pro

### First occurred

today (30th May)

### Steps to reproduce

1. Open the app
2. Get "Error loading posts" error

### Example post, link, markdown...

-

### Expected behaviour

Posts load

### Current behaviour

Throws an error

### Logs

From logcat:
```org.json.JSONException:` Value <!DOCTYPE of type java.lang.String cannot be converted to JSONObject
	at org.json.JSON.typeMismatch(JSON.java:112)
	at org.json.JSONObject.<init>(JSONObject.java:172)
	at org.json.JSONObject.<init>(JSONObject.java:185)```

Likely Reddit broke something in the API again...

**Comment by AlexJoaquimPereira** [comment]
Yeah same with me, it's related to the "posts not loading" thing I raised an issue yesterday. Because the only thing which works is the posts on a profile.

**Comment by bbaster** [comment]
Doesn't seem related to the newest bugfix, this happened too on an older version.

**Comment by ditydot** [comment]
Same for me, but on 7.2.2 beta 1. I was halfway through reading a comment when the post suddenly reloaded, and then a whole lotta nothin'.

**Comment by AlexJoaquimPereira** [comment]
The latest bugfix is the actual solution, we need to apply to the whole app. That isn't the issue. And the app did start crashing out of nowhere then.

If you search for posts they work, but the app crashes in a short while.

**Comment by marovira** [comment]
The issue also occurs on Boost (patched with ReVanced) so it's not just Infinity having issues. Feels like reddit broke something again.

**Comment by Toastie** [comment]
> The latest bugfix is the actual solution, we need to apply to the whole app. That isn't the issue. And the app did start crashing out of nowhere then.
> 
> If you search for posts they work, but the app crashes in a short while.

Yeah, it seems to be related, nothing except my own profile and my saved posts is loading on 7.2.2.-Beta 1

**Comment by yuukiAme** [comment]
v7.2.3 is out now. But it fails to patch with Revanced Manager v1.20.1

`Spoof client failed: app.revanced.patcher.patch.PatchException: Failed to resolve APIUtilsFinger`

so the output app doesn't work with ClientID when I tried to login with my Reddit account.

**Comment by louis-pre** [comment]
The workaround v7.2.3 introduces is to comment out the spoofed client so if Revanced Manager tries to update it (which I'm pretty sure it does) then it will fail. 

You can as easily build your own apk with your own API key, there is a Google Colab for that somewhere or you could do it locally but it's slightly more involved. No need for Revanced here.

**Comment by flokol120** [comment]
@yuukiAme: https://colab.research.google.com/drive/13AE8RvjnCfuBJGaACEqxeBIMo33_l-Sc?usp=sharing

**Comment by 4rtemis-4rrow** [comment]
same issue here

7.1.1, patched with revanced

**Comment by edgan** [comment]
> It was working till about 17h GMT, i was using the app without issues, then stopped working.
> 
> Anonymous works.
> 
> Tested RedReader with logged user, also working.
> 
> Infinity for Reddit: v7.2.2-beta1

Can you say more on RedReader working for you? RedReader seems to work, as a logged in user, mostly if I don't put in a custom access token. Yet some features are broken. If I do put in a custom access token I get errors on logging in.

**Comment by yuukiAme** [comment]
@flokol120 

Thank you. After I changed my `redirect uri` in Reddit preference, everything works just fine after building the APK for the first time ever.

**Comment by edgan** [comment]
> Thank you. After I changed my `redirect uri` in Reddit preference, everything works just fine after building the APK for the first time ever.
> 
> but the version that I built but I did not select was version `7.2.2-beta1`. In the Google Scholar, there was no choosing version `7.2.3`.


@yuukiAme I too got it working.

**Comment by edgan** [comment]
The solution seems to be as seem as removing "android:" from the user agent for Slide, and it seems the same will work for Infinity for Reddit.

```
diff --git a/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java b/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
index 4a87af1c..fb22d72b 100644
--- a/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
+++ b/app/src/main/java/ml/docilealligator/infinityforreddit/utils/APIUtils.java
@@ -46,7 +46,7 @@ public class APIUtils {
     public static final String AUTHORIZATION_KEY = "Authorization";
     public static final String AUTHORIZATION_BASE = "bearer ";
     public static final String USER_AGENT_KEY = "User-Agent";
-    public static final String USER_AGENT = "android:ml.docilealligator.infinityforreddit:v7.2.0-beta1 (by /u/Hostilenemy)";
+    public static final String USER_AGENT = "ml.docilealligator.infinityforreddit:v7.2.0-beta1 (by /u/Hostilenemy)";
 
     public static final String GRANT_TYPE_KEY = "grant_type";
     public static final String GRANT_TYPE_REFRESH_TOKEN = "refresh_token";
```

**Comment by Devitoo** [comment]
so was working 7.2.1 revanced yesterday stopped loading all posts
tried all other versions 7.2.2 beta , 7.2.3 cant patch on revanced 
tried 7.1.1again patched but can not login gives 
There was an error trying to connect with your reddit account
You got here because a third-party application wanted to authorize connecting with your reddit account. However an unknown error occurred, please see details below. Please try refreshing the page or coming back later.
Error: Invalid request to Oauth API

btw tried redirect url http://127.0.0.1 and

**Comment by bbaster** [comment]
Now it's all working fine, even on 7.2.1 with the old patches. Maybe Reddit fixed this on their side...

**Comment by 4rtemis-4rrow** [comment]
yup, working for me too, I do suppose it was an issue on reddit's side

I patched 7.2.2, but it didn't work, kept it and went to sleep, when I woke up, it was working, even though I didn't change anything

**Comment by adripo** [comment]
@Docile-Alligator should we revert back the user-agent to the standard `android:...` now that the issue is solved or we keep using `mobile:...` ?
